x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,224 @@
+    method of providing the GCP credentials this integration will use. You can follow these
+    step-by-step instructions to generate the necessary credentials."
+      />
+    </EuiText>
+  </>
+);
+
+const DocsLink = (
+  <EuiText color={'subdued'} size="s">
+    <FormattedMessage
+      id="xpack.csp.eksIntegration.docsLink"
+      defaultMessage="Read the {docs} for more details"
+      values={{
+        docs: (
+          <EuiLink
+            href="https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html"
+            external
+          >
+            documentation
+          </EuiLink>
+        ),
+      }}
+    />
+  </EuiText>
+);
+
+interface GcpFields {
+  fields: Record<string, { label: string; type?: 'password' | 'text' }>;
+}
+const gcpField: GcpFields = {
+  fields: {
+    project_id: {
+      label: i18n.translate('xpack.csp.gcpIntegration.projectidLabel', {
+        defaultMessage: 'Project ID',
+      }),
+    },
+    credentials_file: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsLabel', {
+        defaultMessage: 'Credentials File',
+      }),
+    },
+    credentials_json: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsjsonLabel', {
+        defaultMessage: 'Credentials JSON',
+      }),
+    },
+  },
+};
+
+const jsonOptionsHolder = [
+  { label: 'Option A', text: 'Option A' },
+  { label: 'Option B', text: 'Option B' },
+];
+
+const getSetupFormatOptions = (): Array<{
+  id: SetupFormatGCP;
+  label: string;
+  disabled: boolean;
+}> => [
+  {
+    id: 'google_cloud_shell',
+    label: 'Google Cloud Shell',
+    disabled: true,
+  },
+  {
+    id: 'manual',
+    label: i18n.translate('xpack.csp.gcpIntegration.setupFormatOptions.manual', {
+      defaultMessage: 'Manual',
+    }),
+    disabled: false,
+  },
+];
+
+export type AwsCredentialsType = keyof typeof gcpField;
+
+interface Props {
+  newPolicy: NewPackagePolicy;
+  input: Extract<
+    NewPackagePolicyPostureInput,
+    { type: 'cloudbeat/cis_aws' | 'cloudbeat/cis_eks' | 'cloudbeat/cis_gcp' }
+  >;
+  updatePolicy(updatedPolicy: NewPackagePolicy): void;
+}
+
+const getInputVarsFields = (input: NewPackagePolicyInput, fields: GcpFields[keyof GcpFields]) =>
+  Object.entries(input.streams[0].vars || {})
+    .filter(([id]) => id in fields)
+    .map(([id, inputVar]) => {
+      const field = fields[id];
+      return {
+        id,
+        label: field.label,
+        type: field.type || 'text',
+        value: inputVar.value,
+      } as const;
+    });
+
+export const GcpCredentialsForm = ({ input, newPolicy, updatePolicy }: Props) => {
+  // We only have a value for 'aws.credentials.type' once the form has mounted.
+  // On initial render we don't have that value so we default to the first option.
+  // const awsCredentialsType = getAwsCredentialsType(input) || AWS_CREDENTIALS_OPTIONS[0].id;
+  //   const group = gcpField;
```

Comment:
again im not sure, i might be missing something, but there are a lot of AWS references here where i would expect everything to be gcp related